### PR TITLE
Fix Impact tab on ITIL objects; fixes #8837

### LINKED ISSUE #8837: Error when clicking into "Impact Analysis" tab from tickets, problems and changes
**Describe the bug**

An error is thrown into ``php-errors.log`` file and no content from "Impact Analysis" is shown into the ticket page when clicking into "Impact Analysis" after upgrading to GLPI 9.5.4.

**Page(s) URL**

https://glpi.example.com/front/ticket.form.php?id=373

**To reproduce**

1. Go to 'Assistance' > 'Tickets'
2. Click on the Name of a ticket
3. Click into ``Impact Analysis`` tab

**Expected behavior**

I'm not sure how it's shown into tickets, but normally into other assets this tab loads a kind of diagram related to the object. It seems that it's being loaded normally with other assets, like computers. It seems to be shown only with tickets, but I'm not sure.

**Logs**

```
[2021-03-15 11:16:36] glpiphplog.CRITICAL:   *** Uncaught Exception TypeError: Argument 1 passed to Impact::isEnabled() must be of the type string, null given, called in /var/www/clients/client0/web1/web/inc/impact.class.php on line 812 in /var/www/clients/client0/web1/web/inc/impact.class.php at line 1725
  Backtrace :
  inc/impact.class.php:812                           Impact::isEnabled()
  inc/impact.class.php:176                           Impact::printAssetSelectionForm()
  inc/commonglpi.class.php:637                       Impact::displayTabContentForItem()
  ajax/common.tabs.php:101                           CommonGLPI::displayStandardTab()
```

**Screenshots**

glpi-impact-analysis-error

**Your GLPI setup (you can find it in Setup > General menu, System tab)**

```
[code]
 
GLPI 9.5.4 ( => /var/www/clients/client0/web1/web)
Installation mode: TARBALL

Server
 
Operating system: Linux ispconfig.example.com 4.18.0-240.10.1.el8_3.x86_64 #1 SMP Mon Jan 18 17:05:51 UTC 2021 x86_64
PHP 7.2.34 fpm-fcgi (Core, PDO, PDO_ODBC, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apcu, bcmath, bz2, calendar, cgi-fcgi,
	ctype, curl, date, dom, exif, fileinfo, filter, ftp, gd, gettext, hash, iconv, imap, intl, json, ldap, libxml, mbstring, mcrypt,
	mysql, mysqli, mysqlnd, odbc, openssl, pcre, pdo_mysql, pdo_sqlite, posix, session, shmop, snmp, soap, sockets, sodium, sqlite3,
	standard, sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xml, xmlreader, xmlrpc, xmlwriter, xsl, zip, zlib)
Setup: max_execution_time="600" memory_limit="128M" post_max_size="150 M" safe_mode="" session.save_handler="files"
	upload_max_filesize="150M" 
Software: Apache ()
	Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.190 Safari/537.36
Server Software: MariaDB Server
	Server Version: 10.3.27-MariaDB
	Server SQL Mode: 
	Parameters: c0glpiuser@localhost/c0glpi
	Host info: Localhost via UNIX socket
	
PHP version is at least 7.2.0 - Perfect!PHP version is at least 7.2.0 - Perfect!
Sessions support is available - Perfect!Sessions support is available - Perfect!
Allocated memory > 64 Mio - Perfect!Allocated memory > 64 Mio - Perfect!
mysqli extension is installedmysqli extension is installed
ctype extension is installedctype extension is installed
fileinfo extension is installedfileinfo extension is installed
json extension is installedjson extension is installed
mbstring extension is installedmbstring extension is installed
iconv extension is installediconv extension is installed
zlib extension is installedzlib extension is installed
curl extension is installedcurl extension is installed
gd extension is installedgd extension is installed
simplexml extension is installedsimplexml extension is installed
intl extension is installedintl extension is installed
ldap extension is installedldap extension is installed
apcu extension is installedapcu extension is installed
Zend OPcache extension is installedZend OPcache extension is installed
xmlrpc extension is installedxmlrpc extension is installed
CAS extension is installedCAS extension is installed
exif extension is installedexif extension is installed
zip extension is installedzip extension is installed
bz2 extension is installedbz2 extension is installed
sodium extension is installedsodium extension is installed
Database version seems correct (10.3.27) - Perfect!Database version seems correct (10.3.27) - Perfect!
Timezones seems loaded in databaseTimezones seems loaded in database
The log file has been created successfully.The log file has been created successfully.
Write access to /var/www/clients/client0/web1/web/files/_cache has been validated.Write access to /var/www/clients/client0/web1/web/files/_cache has been validated.
Write access to /var/www/clients/client0/web1/web/config has been validated.Write access to /var/www/clients/client0/web1/web/config has been validated.
Write access to /var/www/clients/client0/web1/web/files/_cron has been validated.Write access to /var/www/clients/client0/web1/web/files/_cron has been validated.
Write access to /var/www/clients/client0/web1/web/files has been validated.Write access to /var/www/clients/client0/web1/web/files has been validated.
Write access to /var/www/clients/client0/web1/web/files/_dumps has been validated.Write access to /var/www/clients/client0/web1/web/files/_dumps has been validated.
Write access to /var/www/clients/client0/web1/web/files/_graphs has been validated.Write access to /var/www/clients/client0/web1/web/files/_graphs has been validated.
Write access to /var/www/clients/client0/web1/web/files/_lock has been validated.Write access to /var/www/clients/client0/web1/web/files/_lock has been validated.
Write access to /var/www/clients/client0/web1/web/files/_pictures has been validated.Write access to /var/www/clients/client0/web1/web/files/_pictures has been validated.
Write access to /var/www/clients/client0/web1/web/files/_plugins has been validated.Write access to /var/www/clients/client0/web1/web/files/_plugins has been validated.
Write access to /var/www/clients/client0/web1/web/files/_rss has been validated.Write access to /var/www/clients/client0/web1/web/files/_rss has been validated.
Write access to /var/www/clients/client0/web1/web/files/_sessions has been validated.Write access to /var/www/clients/client0/web1/web/files/_sessions has been validated.
Write access to /var/www/clients/client0/web1/web/files/_tmp has been validated.Write access to /var/www/clients/client0/web1/web/files/_tmp has been validated.
Write access to /var/www/clients/client0/web1/web/files/_uploads has been validated.Write access to /var/www/clients/client0/web1/web/files/_uploads has been validated.
Write access to /var/www/clients/client0/web1/web/marketplace has been validated.Write access to /var/www/clients/client0/web1/web/marketplace has been validated.
Web access to files directory is protectedWeb access to files directory is protected
For security reasons, SELinux mode should be Enforcing.For security reasons, SELinux mode should be Enforcing.

GLPI constants
 
GLPI_ROOT: /var/www/clients/client0/web1/web
GLPI_CONFIG_DIR: /var/www/clients/client0/web1/web/config
GLPI_VAR_DIR: /var/www/clients/client0/web1/web/files
GLPI_MARKETPLACE_DIR: /var/www/clients/client0/web1/web/marketplace
GLPI_USE_CSRF_CHECK: 1
GLPI_CSRF_EXPIRES: 7200
GLPI_CSRF_MAX_TOKENS: 100
GLPI_USE_IDOR_CHECK: 1
GLPI_IDOR_EXPIRES: 7200
GLPI_ALLOW_IFRAME_IN_RICH_TEXT: 
GLPI_TELEMETRY_URI: https://telemetry.glpi-project.org
GLPI_INSTALL_MODE: TARBALL
GLPI_NETWORK_MAIL: glpi@teclib.com
GLPI_NETWORK_SERVICES: https://services.glpi-network.com
GLPI_MARKETPLACE_PRERELEASES: 
GLPI_USER_AGENT_EXTRA_COMMENTS: 
GLPI_AJAX_DASHBOARD: 1
GLPI_CALDAV_IMPORT_STATE: 0
GLPI_DEMO_MODE: 0
GLPI_FORCE_EMPTY_SQL_MODE: 1
GLPI_DOC_DIR: /var/www/clients/client0/web1/web/files
GLPI_CACHE_DIR: /var/www/clients/client0/web1/web/files/_cache
GLPI_CRON_DIR: /var/www/clients/client0/web1/web/files/_cron
GLPI_DUMP_DIR: /var/www/clients/client0/web1/web/files/_dumps
GLPI_GRAPH_DIR: /var/www/clients/client0/web1/web/files/_graphs
GLPI_LOCAL_I18N_DIR: /var/www/clients/client0/web1/web/files/_locales
GLPI_LOCK_DIR: /var/www/clients/client0/web1/web/files/_lock
GLPI_LOG_DIR: /var/www/clients/client0/web1/web/files/_log
GLPI_PICTURE_DIR: /var/www/clients/client0/web1/web/files/_pictures
GLPI_PLUGIN_DOC_DIR: /var/www/clients/client0/web1/web/files/_plugins
GLPI_RSS_DIR: /var/www/clients/client0/web1/web/files/_rss
GLPI_SESSION_DIR: /var/www/clients/client0/web1/web/files/_sessions
GLPI_TMP_DIR: /var/www/clients/client0/web1/web/files/_tmp
GLPI_UPLOAD_DIR: /var/www/clients/client0/web1/web/files/_uploads
GLPI_NETWORK_REGISTRATION_API_URL: https://services.glpi-network.com/api/registration/
GLPI_MARKETPLACE_PLUGINS_API_URI: https://services.glpi-network.com/api/glpi-plugins/
GLPI_I18N_DIR: /var/www/clients/client0/web1/web/locales
GLPI_VERSION: 9.5.4
GLPI_SCHEMA_VERSION: 9.5.4
GLPI_MIN_PHP: 7.2.0
GLPI_YEAR: 2021

Libraries
 
htmlawed/htmlawed version 1.2.5 in (/var/www/clients/client0/web1/web/vendor/htmlawed/htmlawed)
phpmailer/phpmailer version 6.1.6 in (/var/www/clients/client0/web1/web/vendor/phpmailer/phpmailer/src)
simplepie/simplepie version 1.5.6 in (/var/www/clients/client0/web1/web/vendor/simplepie/simplepie/library)
tecnickcom/tcpdf version 6.3.5 in (/var/www/clients/client0/web1/web/vendor/tecnickcom/tcpdf)
michelf/php-markdown in (/var/www/clients/client0/web1/web/vendor/michelf/php-markdown/Michelf)
true/punycode in (/var/www/clients/client0/web1/web/vendor/true/punycode/src)
iamcal/lib_autolink in (/var/www/clients/client0/web1/web/vendor/iamcal/lib_autolink)
sabre/dav in (/var/www/clients/client0/web1/web/vendor/sabre/dav/lib/DAV)
sabre/http in (/var/www/clients/client0/web1/web/vendor/sabre/http/lib)
sabre/uri in (/var/www/clients/client0/web1/web/vendor/sabre/uri/lib)
sabre/vobject in (/var/www/clients/client0/web1/web/vendor/sabre/vobject/lib)
laminas/laminas-cache in (/var/www/clients/client0/web1/web/vendor/laminas/laminas-cache/src)
laminas/laminas-i18n in (/var/www/clients/client0/web1/web/vendor/laminas/laminas-i18n/src)
laminas/laminas-serializer in (/var/www/clients/client0/web1/web/vendor/laminas/laminas-serializer/src)
monolog/monolog in (/var/www/clients/client0/web1/web/vendor/monolog/monolog/src/Monolog)
sebastian/diff in (/var/www/clients/client0/web1/web/vendor/sebastian/diff/src)
elvanto/litemoji in (/var/www/clients/client0/web1/web/vendor/elvanto/litemoji/src)
symfony/console in (/var/www/clients/client0/web1/web/vendor/symfony/console)
scssphp/scssphp in (/var/www/clients/client0/web1/web/vendor/scssphp/scssphp/src)
laminas/laminas-mail in (/var/www/clients/client0/web1/web/vendor/laminas/laminas-mail/src/Protocol)
laminas/laminas-mime in (/var/www/clients/client0/web1/web/vendor/laminas/laminas-mime/src)
rlanvin/php-rrule in (/var/www/clients/client0/web1/web/vendor/rlanvin/php-rrule/src)
blueimp/jquery-file-upload in (/var/www/clients/client0/web1/web/vendor/blueimp/jquery-file-upload/server/php)
ramsey/uuid in (/var/www/clients/client0/web1/web/vendor/ramsey/uuid/src)
psr/log in (/var/www/clients/client0/web1/web/vendor/psr/log/Psr/Log)
psr/simple-cache in (/var/www/clients/client0/web1/web/vendor/psr/simple-cache/src)
mexitek/phpcolors in (/var/www/clients/client0/web1/web/vendor/mexitek/phpcolors/src/Mexitek/PHPColors)
guzzlehttp/guzzle in (/var/www/clients/client0/web1/web/vendor/guzzlehttp/guzzle/src)
guzzlehttp/psr7 in (/var/www/clients/client0/web1/web/vendor/guzzlehttp/psr7/src)
wapmorgan/unified-archive in (/var/www/clients/client0/web1/web/vendor/wapmorgan/unified-archive/src)
paragonie/sodium_compat in (/var/www/clients/client0/web1/web/vendor/paragonie/sodium_compat/src)
phpCas version 1.3.8 in (/usr/share/pear)

LDAP directories
 
Server: 'ldaps://example.com', Port: '389', BaseDN: 'OU=example,DC=example,DC=com,DC=br', Connection filter:
		'(&(objectClass=user)(objectCategory=person)(!(userAccountControl:1.2.840.113556.1.4.803:=2))(!(memberOf=CN=Contas de Serviço -
		Negar Logon Interativo,OU=Serviços,OU=TI,OU=example,DC=example,DC=com,DC=br))(!(sAMAccountName=fabrica))(!(sAMAccountName=cq))(!(sAMAccountName=manutencao)))',
		RootDN: 'glpiaduser@example.com', Use TLS: none

SQL replicas
 
Not active

Notifications
 
Way of sending emails: SMTP+SSL (glpi@example.com@smtp.kinghost.net)

Mails receivers
 
Name: 'glpi@example.com' Active: Yes
	Server: '{smtp.kinghost.net:995/pop/ssl/validate-cert}' Login: 'glpi@example.com' Password: Yes

Plugins list
 
	fields               Name: Campos adicionais              Version: 1.12.2     State: Enabled
	behaviors            Name: Comportamentos                 Version: 2.4.1      State: Enabled
	accounts             Name: Contas                         Version: 2.6.0      State: Enabled
	archimap             Name: Diagrams                       Version: 2.2.1      State: Enabled
	fusioninventory      Name: FusionInventory                Version: 9.5+2.0  State: Enabled
	genericobject        Name: Gerenciamento de objetos       Version: 2.10.0     State: Installed / not activated
	manufacturersimports Name: Importação de fornecedores     Version: 2.3.1      State: Enabled
	pdf                  Name: Imprimir em PDF                Version: 1.7.0      State: Enabled
	mreporting           Name: Mais Relatórios                Version: 1.7.1      State: Enabled
	printercounters      Name: Printer counters               Version: 1.7.0      State: Enabled
	reports              Name: Relatórios                     Version: 1.14.0     State: Enabled
	taskdrop             Name: TaskDrop                       Version: 1.2.0      State: Enabled

[/code]
```

**Additional context**

None.

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #8837

### CODE REVIEW

**Review by orthagh — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Review by yllen — APPROVED** [review]

File: inc/impact.class.php [hunk 1]
```diff
@@ -156,24 +156,30 @@ public static function displayTabContentForItem(
...
-                  $found = $item->getFromDB($linked_item_id);
-                  break 2;
+                  if ($item->getFromDB($linked_item_id)) {
+                     $items_data[] = [
+                        'itemtype' => $itemtype,
+                        'items_id' => $linked_item_id,
+                        'name'     => $item->getNameID(),
+                     ];
+                  }
+                  continue;
```
Comment (AdrienClairembault) [inline inc/impact.class.php:172]: `continue` statement doesn't seem needed since this is already the end of the loop.

**Review by AdrienClairembault — APPROVED** [review]

File: inc/impact.class.php (hunk 1 above)
Comment (cedric-anne) [inline inc/impact.class.php:172]:
```suggestion
```

Comment (cedric-anne) [inline inc/impact.class.php:172]:
```suggestion
                  if (!$item->getFromDB($linked_item_id)) {
                     continue;
                  }
                  $items_data[] = [
                     'itemtype' => $itemtype,
                     'items_id' => $linked_item_id,
                     'name'     => $item->getNameID(),
                  ];
```

Comment (cedric-anne) [reply inc/impact.class.php]: I reorganized code.

**Review by AdrienClairembault — APPROVED** [review]

**Pull request merged**